Sources/Subs.php

```diff
@@ -4308,7 +4308,7 @@ function template_javascript($do_deferred = false)
 
 		foreach ($context['javascript_vars'] as $key => $value)
 		{
-			if (empty($value))
+			if (empty($value) && !is_numeric($value))
```

Comment:
What if the var should be initialized as an empty string, array, object? Would `null` suffice here?
```suggestion
			if ($value === null)
```

Reply:
I don't think we can handle arrays and objects here in any case, but I suppose initiating as an empty string makes sense. For that matter, initiating as false also makes sense.

Reply:
As a side issue, it looks like we don't currently support adding empty values except '0' in `addJavaScriptVar()`. I don't see any reason for that limitation, though. A mod might very well want to initiate a var with an empty string or an explicit `false` or even as an empty array. Perhaps we should expand support for those types of empty variables in `addJavaScriptVar()`.

Either way, though, it still makes sense to improve our handling of various empty values here. After all, it is entirely possible for a mod to bypass `addJavaScriptVar()` and its limitations by appending directly to `$context['javascript_vars']`.